### PR TITLE
doc: debounceAsyncResult

### PR DESCRIPTION
**Pull request by bowencool**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  update select-users demo  |
| 🇨🇳 Chinese |  更新搜索示例  |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/select/demo/select-users.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ A complete multiple select sample with remote search, debounce fetch, ajax callb
 import { Select, Spin } from 'antd';
 import type { SelectProps } from 'antd/es/select';
 import debounce from 'lodash/debounce';
+import { debounceAsyncResult } from 'high-order-async-utilities';
```
Comment (afc163) [inline components/select/demo/select-users.md:20]: Any chance to use lodash?

Comment (bowencool) [reply components/select/demo/select-users.md:20]: I didn't find a similar API in lodash or even in the open source community, so I released this package myself

Comment (bowencool) [reply components/select/demo/select-users.md:20]: The difference between debounceAsyncResult and debounce are: debounce handles the timing of task execution. debounceAsyncResult handles the result of an asynchronous task that has already been executed.

debounceAsyncResult 和 debounce 的区别是： debounce 处理任务的执行时机。 debounceAsyncResult 处理已经执行的异步任务的结果。

Comment (ntvinhit) [reply components/select/demo/select-users.md:20]: I know `high-order-async-utilities` is your library, but I don't think it's a good idea if we put them here

Comment (bowencool) [reply components/select/demo/select-users.md:20]: > I know `high-order-async-utilities` is your library, but I don't think it's a good idea if we put them here

What's the reason?

**Comment by afc163** [comment]
抱歉，demo 代码暂时不会依赖一个不成熟的库。

**Comment by bowencool** [comment]
怎么定义成熟，star数？知名度？我能做的都已经做了，demo 全功能演示，文档、更新日志写的都很清晰，单测覆盖率也满了。你要说功能不足或者作用不大，我还能接受，这个需要慢慢迭代。你要说因为用的人不多，所以不成熟，我就觉得有点“何不食肉糜”的意思了。 2022年9月19日 +0800 13:42 afc163 ***@***.***>，写道： 抱歉，demo 代码暂时不

**Comment by afc163** [comment]
> 你要说功能不足或者作用不大
我反而相信你的功能和作用都能满足这个场景的需求。但是 antd demo 会被业务拷贝走，我们希望尽可能清晰和减少黑盒，这样业务代码遇到问题也可以自行排查和解决。如果有外部依赖也希望是社区广泛流行的库，或者是官方自行维护的库。

**Comment by bowencool** [comment]
> > 你要说功能不足或者作用不大
> 
> 我反而相信你这个库的功能和作用都能满足这个场景的需求。但是 antd demo 会被业务的开发者拷贝代码做二次开发，我们希望尽可能清晰和减少黑盒，这样业务代码遇到问题也可以自行排查和解决。如果有外部依赖也希望是社区广泛流行的库，或者是官方自行维护的库。

了解了

**Pull request closed without merging**